includes/sanitizers/class-amp-img-sanitizer.php

```diff
@@ -552,4 +567,19 @@ private function is_gif_url( $url ) {
 		$path = wp_parse_url( $url, PHP_URL_PATH );
 		return substr( $path, -strlen( $ext ) ) === $ext;
 	}
+
+	/**
+	 * Determines if a URL is a known tracking pixel URL.
+	 *
+	 * Currently, only Facebook tracking pixel URL is detected.
+	 *
+	 * @since 2.2.2
+	 *
+	 * @param string $url URL to inspect.
+	 *
+	 * @return bool Returns true if $url is a tracking pixel URL.
+	 */
+	private static function is_tracking_pixel_url( $url ) {
+		return (bool) preg_match( '/http(?>s)?:\/\/(?>www\.)?facebook\.com\/tr\?/i', $url );
```

Comment:
Maybe parsing the URL would be a bit easier to work with?

```suggestion
		$parsed_url = wp_parse_url( $url );
		return isset( $parsed_url['host'], $parsed_url['path'] ) && 'facebook.com' === str_replace( 'www.',  $parsed_url['host'] ) && '/tr' === $parsed_url['path'];
```